### PR TITLE
Openmetrics Parser

### PR DESCRIPTION
**Pull request by brian-brazil**
Based on https://docs.google.com/document/u/1/d/1KwV0mAXwwbvvifBvDKH_LU1YjyXE_wxCkHNoCGq1GX0/edit

### CODE REVIEW

**Comment by roidelapluie** [comment]
Did I miss something? I still can not find exactly the open metrics standard defined properly somewhere. Feels strange to start the implementation before we have that.

**Comment by RichiH** [comment]
This is blocking on me, unfortunately.

Is this a general comment or a hard blocker to get things in as `EXPERIMENTAL`?

**Comment by brian-brazil** [comment]
I don't see this as a hard blocker, this will actually help us flesh things out.

**Comment by roidelapluie** [comment]
It is not a hard blocker, I was just wondering on which "draft'  this PR is based :)

File: Makefile [hunk 1]
```diff
@@ -18,6 +18,7 @@ STATICCHECK_IGNORE = \
   github.com/prometheus/prometheus/discovery/kubernetes/node.go:SA1019 \
   github.com/prometheus/prometheus/documentation/examples/remote_storage/remote_storage_adapter/main.go:SA1019 \
   github.com/prometheus/prometheus/pkg/textparse/promlex.l.go:SA4006 \
+  github.com/prometheus/prometheus/pkg/textparse/omlex.l.go:SA4006 \
```
[the remark this answers is not in the document]
Comment (brian-brazil) [reply Makefile]: Done

File: pkg/textparse/interface.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 The Prometheus Authors
```
Comment (SuperQ) [inline pkg/textparse/interface.go]: 2018

File: pkg/textparse/openmetricslex.l [hunk 3]
```diff
@@ -0,0 +1,71 @@
+%{
+// Copyright 2017 The Prometheus Authors
```
Comment (SuperQ) [inline pkg/textparse/openmetricslex.l]: 2018

File: pkg/textparse/openmetricsparse.go [hunk 4]
```diff
@@ -0,0 +1,346 @@
+// Copyright 2017 The Prometheus Authors
```
Comment (SuperQ) [inline pkg/textparse/openmetricsparse.go]: 2018

File: pkg/textparse/interface.go (hunk 2 above)
Comment (brian-brazil) [reply pkg/textparse/interface.go]: All done

**Pull request merged**